### PR TITLE
Generate OTP Endpoint

### PR DESCRIPTION
**Pull request by zandre-eng**
## Technical Summary

Link to ticket [here](https://dimagi.atlassian.net/browse/CCCT-1521).
Link to tech spec [here](https://docs.google.com/document/d/1hhHsbgD6PT_bpxp6ZnyyGZbQcHCV7MUjJJG1bGUubOE/edit?tab=t.0#heading=h.hp2snhwp807v).

A new GET API endpoint is added which allows for generating a new OTP for a specific `SessionPhoneDevice` instance. Specifically, the most recent `SessionPhoneDevice` instance for a phone number is fetched, as they may be more than one over a phone number's lifetime.

**Endpoint Overview:**
Input: 
The endpoint takes a `phone_number` string query param as input.

Output:
On success the endpoint gives the following JSON output:
```
{"otp": <session_otp>}
```

If no phone number is given then the following output is given with a status code of 400:
```
{"error_code": "MISSING_DATA"}
```

Similarly, if the given phone number doesn't match any user session, or if the matched user session is expired then the following output is given with a status code of 404:
```
{"error_code": "SESSION_NOT_FOUND"}
```

Please note, once the `ze/flag-for-manual-otp` branch has been merged, I'll change this PR to merge into main. I'm using `ze/flag-for-manual-otp` as the base branch for now since this PR relies on changes made in that branch.

## Logging and monitoring

No logging requirements.

### Safety story

- Local testing
- Unit tests

The code introduced in this PR is all new and does not touch on any existing functionality.

- [X] I am confident that this change will not break current and/or previous versions of CommCare apps

### Automated test coverage

Automated tests exist to ensure that the happy path for the new API endpoint acts as expected, as well as testing various edge cases.

### QA Plan

No QA planned.

### Labels & Review

- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by Charl1996 — APPROVED** [review]

**Review by calellowitz — COMMENTED** [review]
One question, but the rest seems fine

File: users/views.py [hunk 1]
```diff
@@ -912,3 +912,28 @@ def report_integrity(request):
...
+            return JsonResponse({"error_code": ErrorCodes.MISSING_DATA}, status=400)
+
+        session_phone_device = (
+            SessionPhoneDevice.objects.filter(phone_number=phone_number).order_by("-session__created").first()
+        )
+        if not (session_phone_device and session_phone_device.session.is_valid()):
+            return JsonResponse({"error_code": ErrorCodes.SESSION_NOT_FOUND}, status=404)
+
+        expire_seconds = 60 * 60 * 4  # Four hours
+        session_phone_device.generate_token(valid_secs=expire_seconds)
```
[the remark this answers is not in the document]
Comment (zandre-eng) [reply users/views.py]: [50666a3](https://github.com/dimagi/connect-id/pull/163/commits/50666a3ccf89705e0f9e989f3d17b74eca5e1bce)

**Review by Charl1996 — APPROVED** [review]
